headers/fmi3FunctionTypes.h

```diff
@@ -173,14 +120,63 @@ typedef fmi3Status  fmi3SetDebugLoggingTYPE(fmi3Instance instance,
 
 /* Creation and destruction of FMU instances and setting debug status */
 /* tag::Instantiate[] */
-typedef fmi3Instance fmi3InstantiateTYPE(fmi3String        instanceName,
-                                         fmi3InterfaceType fmuType,
-                                         fmi3String        fmuInstantiationToken,
-                                         fmi3String        fmuResourceLocation,
-                                         const fmi3CallbackFunctions* functions,
-                                         fmi3Boolean       visible,
-                                         fmi3Boolean       loggingOn,
-                                         const fmi3CoSimulationConfiguration* fmuCoSimulationConfiguration);
+typedef fmi3Instance fmi3InstantiateModelExchangeTYPE(
+    fmi3String                 instanceName,
+    fmi3String                 instantiationToken,
+    fmi3String                 resourceLocation,
+    fmi3Boolean                visible,
+    fmi3Boolean                loggingOn,
+    fmi3InstanceEnvironment    instanceEnvironment,
+    fmi3CallbackLogMessage     logMessage,
+    fmi3CallbackAllocateMemory allocateMemory,
+    fmi3CallbackFreeMemory     freeMemory);
+
+typedef fmi3Instance fmi3InstantiateBasicCoSimulationTYPE(
+    fmi3String                     instanceName,
+    fmi3String                     instantiationToken,
+    fmi3String                     resourceLocation,
+    fmi3Boolean                    visible,
+    fmi3Boolean                    loggingOn,
+    fmi3Boolean                    intermediateVariableGetRequired,
+    fmi3Boolean                    intermediateInternalVariableGetRequired,
+    fmi3Boolean                    intermediateVariableSetRequired,
```

Comment:
Is intermediate access supported by basic co-simulation?

Reply:
I would go further: Should Basic Co-Simulation support Intermediate Variable Access?
Or: should IAV be allowed only for Hybrid CS?

Reply:
See related discussions in issue #820 and #797.

Reply:
> I would go further: Should Basic Co-Simulation support Intermediate Variable Access?
> Or: should IAV be allowed only for Hybrid CS?

I moved this discussion regarding IVA and Early return (from #797, #820, #821) to #836.